schematic/store/synapse.py

```diff
@@ -1423,11 +1527,18 @@ def buildDB(
                 manifest_table_id = tableOps.updateTable()
 
         if table_manipulation and table_manipulation.lower() == "upsert":
-            existing_tables = self.get_table_info(datasetId=datasetId)
-            tableId = existing_tables[table_name]
-            annos = self.syn.get_annotations(tableId)
+            table_entity = self.synapse_entity_tracker.get(
+                synapse_id=existing_table_id, syn=self.syn, download_file=False
+            )
+            annos = OldAnnotations(
+                id=table_entity.id,
+                etag=table_entity.etag,
+                values=table_entity.annotations,
+            )
             annos["primary_key"] = table_manifest["Component"][0] + "_id"
             annos = self.syn.set_annotations(annos)
+            table_entity.etag = annos.etag
+            table_entity.annotations = annos
```

Comment:
In this file 2 sets of Annotations from SYNPY are being used:
```
from synapseclient.models.annotations import Annotations # This is the new set of Annotations that has Async support
from synapseclient import Annotations as OldAnnotations # This is the old set of Annotations that works with the current Entity objects
```
Since all of the Entities returned from `syn.get()` still use `OldAnnotations` I am using it in this context.